supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala

```diff
@@ -47,8 +51,56 @@ sealed trait ConfigAsyncJobExecutionActor extends SharedFileSystemAsyncJobExecut
   def writeTaskScript(script: Path, taskName: String, inputs: WorkflowCoercedInputs): Unit = {
     val task = configInitializationData.wdlNamespace.findTask(taskName).
       getOrElse(throw new RuntimeException(s"Unable to find task $taskName"))
-    val inputsWithFqns = inputs map { case (k, v) => s"$taskName.$k" -> v }
-    val command = task.instantiateCommand(task.inputsFromMap(inputsWithFqns), NoFunctions).get
+
+    val taskDefinition = task.womDefinition.toTry.get
+
+    // Build WOM versions of the provided inputs by correlating with the `InputDefinition`s on the `TaskDefinition`.
+    val providedWomInputs: WomEvaluatedCallInputs = (for {
+      input <- inputs.toList
+      (localName, womValue) = input
+      womInputDefinition <- taskDefinition.inputs
+      if localName == womInputDefinition.localName.value
+    } yield womInputDefinition -> womValue).toMap
+
+    /*
+     * TODO WOM FIXME #2946
+     * This forces `none`s onto potentially initialized optionals. This is currently required because otherwise
+     * expressions using uninitialized optionals explode on evaluation. e.g.:
+     *
+     * runtime-attributes = """
+     * String? docker
+     * String? docker_user
+     * """
+     * submit = "/bin/bash ${script}"
+     * submit-docker = """
+     * docker run \
+     *   --cidfile ${docker_cid} \
+     *   --rm -i \
+     *   ${"--user " + docker_user} \
+     *   --entrypoint /bin/bash \
+     *   -v ${cwd}:${docker_cwd} \
+     *   ${docker} ${script}
+     *  """
+     *
+     * The `docker_user` variable is an uninitialized optional. But the expression `${"--user " + docker_user}` will
+     * fail to evaluate if `docker_user` is not provided as an input by the caller of `TaskDefinition#instantiateCommand`.
+     * This only appears to be the case when calling `instantiateCommand` on a `TaskDefinition` that came from
+     * `WdlTask#womDefinition`, which makes it seem that there is some special initialization of task optionals taking
+     * place when create a `WomExecutable` from a `WdlWorkflow`.
```

Comment:
I believe the string interpolation does indeed do some magic for optionals along the lines of "if it doesn't exist, use the `default=` tag, and if that doesn't exist, use `""`"

Reply:
So in this case, I think they might actually not need further "fixing" (as long as set values aren't overridden to `None` too?)

Reply:
Let's discuss this IRL tomorrow.